src/main/service/FirebaseService.ts

```diff
@@ -0,0 +1,31 @@
+import { auth } from 'firebase-admin/lib/auth';
+import Auth = auth.Auth;
+
+/**
+ * Service class for operations related to firebase
+ */
+export class FirebaseService {
+  private admin: Auth;
+
+  constructor(firebaseAdmin: Auth) {
+    this.admin = firebaseAdmin;
+  }
```

Comment:
Benefit of dependency injection: Notice how our controller tests don't need to go about a heavy mess to make sure that `FirebaseService::firebaseAdmin` is not the real `Auth` we will be using in production, instead it will be a mock.